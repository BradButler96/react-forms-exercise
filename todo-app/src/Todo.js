import React, { useState } from "react";
import './todo.css'


const Todo = ({ id, task, edit, remove }) => {

    const [editTodo, setEditTodo] = useState(task)
    const [editing, setEditing] = useState(false)

    const handleDelete = (e) => {
        remove(e.target.previousSibling.previousSibling)
    }

    const handleEdit = (e) => {
        e.preventDefault();
        edit(id, editTodo)
        setEditing(false)
    }

    // Toggle edit input
    const toggleEdit = () => {
        editing ? setEditing(false) : setEditing(true)
    }

    // Update input value when changed
    const handleChange = (e) => {
        setEditTodo(e.target.value)
    }

    let todoContainer;
    
    !editing ? 
    todoContainer = (
        <div className='todo-container' data-testid={`todo`}>
            <div id={ id } className='task' data-testid={`task`}>{ task }</div>
            <button className='edit-btn' data-testid={`edit-btn`} onClick={ toggleEdit }>Edit</button>
            <button className='delete-btn' data-testid={`delete-btn`} onClick={ handleDelete }>X</button>
        </div>
    ) : 
    todoContainer = (
        <div className='todo-container' data-testid={`edit-todo`}>
            <input
                data-testid={`edit-todo-input`}
                type="text"
                value={ editTodo }
                onChange={ handleChange }
            />
            <button className='submit-edit-btn' data-testid={`submit-edit-btn`} onClick={ handleEdit }>Submit </button>
            <button className='cancel-edit-btn' data-testid={`cancel-edit-btn`} onClick={ toggleEdit }>Cancel</button>
        </div>
    )

    return todoContainer
}

export default Todo