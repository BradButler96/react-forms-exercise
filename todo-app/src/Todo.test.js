import React from "react";
import { render } from "@testing-library/react";
import Todo from './Todo.js'

it("renders without crashing", () => {
    render(<Todo id={1} task={'Testing'} />);
});

it("matches snapshot", () => {
    const { asFragment } = render(<Todo id={1} task={'Testing'} />);
    expect(asFragment()).toMatchSnapshot();
});