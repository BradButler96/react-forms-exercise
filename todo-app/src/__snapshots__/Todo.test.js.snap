// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="todo-container"
  >
    <div
      class="task"
      id="1"
    >
      Testing
    </div>
    <button
      class="edit-btn"
    >
      Edit
    </button>
    <button
      class="delete-btn"
    >
      X
    </button>
  </div>
</DocumentFragment>
`;
